### PR TITLE
fix: provider resolution

### PR DESCRIPTION
**Pull request by Jakub-Vacek**
## Description
This PR fixes problem with missing local provider and existing remote provider.

Partially fixes: https://github.com/superfaceai/cli/issues/309 - for new, map and execute command. For prepare command we need to do PR in engine.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have updated the documentation accordingly. For updating Oclif commands documentation use [oclif-dev](https://github.com/oclif/dev-cli#oclif-dev-readme).
- [ ] I have read the **CONTRIBUTION_GUIDE** document.
- [ ] I haven't repeated the code. (DRY)
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by janhalama — COMMENTED** [review]
I went through it. It looks good, I have just two questions. I am not sure if it is important for the user to distinguish between the two locations. They should be synced automatically or?

File: src/common/provider.ts [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+      resolvedProviderJson = assertProviderJson(response.definition);
+    } catch (e) {
+      if (e instanceof AssertionError) {
+        throw userError(`Invalid provider.json. ${e.message}`, 1);
+      }
+      throw userError(`Invalid provider.json - invalid JSON`, 1);
+    }
+  } else if (providerResponse.status === 404) {
+    throw userError(
+      `Provider ${providerName} does not exist both locally (checked ${buildProviderPath(
```
[the remark this answers is not in the document]
Comment (Jakub-Vacek) [reply src/common/provider.ts:121]: Yeah, it is not. I will simplify the message.

File: src/common/provider.ts [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+  ) {
+    if (resolvedProviderJson.source === 'local') {
+      throw userError(
+        `Provider.json file is not properly configured. Please make sure to replace 'TODO' in baseUrl with the actual base url of the API.`,
+        1
+      );
+    }
+
+    throw userError(
+      `Provider.json file downloaded but it is not properly configured. Please make sure to replace 'TODO' in baseUrl with the actual base url of the API.`,
```
[the remark this answers is not in the document]
Comment (Jakub-Vacek) [reply src/common/provider.ts:72]: Yes, it is. I will probably add location of saved file to message.

**Review by kysely — COMMENTED** [review]
Can we please use the full command name `superface` instead of `sf`? It works when installed via Homebrew but I feel like the full name is more clear (despite being more verbose)

**Review by kysely — APPROVED** [review]
The review changes look fine 👍

**Pull request merged**